### ISSUE TITLE
Index Checker `startsWith` refinement

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#5201]
Consider this code (which is in `checker/tests/index/StartsEndsWith.java`):

```java
  final String prefix;

  String propertyName(String methodName) {
    if (methodName.startsWith(prefix)) {
      String result = methodName.substring(prefix.length());
      return result;
    } else {
      return null;
    }
  }
```

The Index Checker issues the following warning:

```
StartsEndsWith.java:17: error: [argument] incompatible argument for parameter beginIndex of substring.
      String result = methodName.substring(prefix.length());
                                                        ^
  found   : @LTEqLengthOf("this.prefix") int
  required: @LTEqLengthOf("methodName") int
```

but the code is correct because `methodName` is at least as long as `this.prefix`.

**Comment by kelloggm** [comment]
This would require the Index Checker to support reasoning about relationships between sequences that are not the same length; i.e., it would require a `@LessThanLen` counter-part to `@SameLen`. I think doing that would require adding another type system, so unless we are seeing this problem commonly in practice it's probably not worth implementing.

That said, implementing a type system for `@LessThanLen` and integrating it into the Index Checker might make a good project for an inexperienced contributor, such as a GSoC student.

**Comment by DhirajSardar** [comment]
Hi I am a newbie .
How can I get started for contributing to this issue?

**Comment by kelloggm** [comment]
Hi! We'd be happy to have your help. If you're not familiar with the Index Checker, we recommend you do a case study using the tool first - that is, use the tool on some code you're familiar with - before you try to start working on the tool's internals. For more details on what that process looks like, you can read this document: https://checkerframework.org/manual/gsoc-ideas.html#get-started
